From ad4e4e466c1e615062d42bff01c1909421e362a4 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Sun, 14 Sep 2025 15:03:05 +0200
Subject: [PATCH 83/95] net: phy: realtek: Perform single PHYCR2 access

The PHYCR2 is currently accessed twice in sequence. Simplify
the code to a single access in preparation for the follow up
patch which would add yet another PHYCR2 access.

Upstream-Status: Pending
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 drivers/net/phy/realtek/realtek_main.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/net/phy/realtek/realtek_main.c b/drivers/net/phy/realtek/realtek_main.c
index 16a347084293e..f65c985de0251 100644
--- a/drivers/net/phy/realtek/realtek_main.c
+++ b/drivers/net/phy/realtek/realtek_main.c
@@ -564,7 +564,7 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 {
 	struct rtl821x_priv *priv = phydev->priv;
 	struct device *dev = &phydev->mdio.dev;
-	u16 val_txdly, val_rxdly;
+	u16 val_txdly, val_rxdly, phycr2_mask;
 	int ret;
 
 	ret = phy_modify_paged_changed(phydev, RTL8211F_PHYCR_PAGE, RTL8211F_PHYCR1,
@@ -637,13 +637,11 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 		return 0;
 
 	/* Disable PHY-mode EEE so LPI is passed to the MAC */
-	ret = phy_modify_paged(phydev, RTL8211F_PHYCR_PAGE, RTL8211F_PHYCR2,
-			       RTL8211F_PHYCR2_PHY_EEE_ENABLE, 0);
-	if (ret)
-		return ret;
+	priv->phycr2 &= ~RTL8211F_PHYCR2_PHY_EEE_ENABLE;
+	phycr2_mask = RTL8211F_PHYCR2_PHY_EEE_ENABLE | RTL8211F_CLKOUT_EN;
 
 	ret = phy_modify_paged(phydev, RTL8211F_PHYCR_PAGE,
-			       RTL8211F_PHYCR2, RTL8211F_CLKOUT_EN,
+			       RTL8211F_PHYCR2, phycr2_mask,
 			       priv->phycr2);
 	if (ret < 0) {
 		dev_err(dev, "clkout configuration failed: %pe\n",
-- 
2.51.0

